Control/lib/services/WorkflowTemplate.service.js

```diff
@@ -59,34 +69,36 @@ class WorkflowTemplateService {
   /**
    * Retrieve a list of mappings for simplified creation of environments based on workflow saved configurations
    * @return {Array<{label: String, configuration: String}>} - list of mappings to be displayed
+   * @throws
    */
   async retrieveWorkflowMappings() {
+    let mappingsString = '';
     try {
-      const mappingsString = await this._apricotGrpc.getRuntimeEntryByComponent(RUNTIME_COMPONENT, RUNTIME_KEY);
-      const mappings = JSON.parse(mappingsString);
-      if (Array.isArray(mappings)) {
-        return mappings.sort(({label: labelA}, {label: labelB}) => labelA < labelB ? -1 : 1);
-      }
-
-      return [];
+      mappingsString = await this._apricotGrpc.getRuntimeEntryByComponent(RUNTIME_COMPONENT, RUNTIME_KEY);
     } catch (error) {
       throw grpcErrorToNativeError(error);
     }
+    const mappings = JSON.parse(mappingsString);
+    if (Array.isArray(mappings)) {
```

Comment:
I personally prefer to explicitely catch specific cases in ifs then continue the process, it avoids nested ifs if you need to add more cases (`if (!Array.isArray(mappings)) {return []}`)

Reply:
You are right, long-term it would make the code easier to read